sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleHistogramAggregator.java

```diff
@@ -0,0 +1,168 @@
+  }
+
+  @Override
+  public final MetricData toMetricData(
+      Map<Labels, HistogramAccumulation> accumulationByLabels,
+      long startEpochNanos,
+      long lastCollectionEpoch,
+      long epochNanos) {
+    return MetricData.createDoubleHistogram(
+        getResource(),
+        getInstrumentationLibraryInfo(),
+        getInstrumentDescriptor().getName(),
+        getInstrumentDescriptor().getDescription(),
+        getInstrumentDescriptor().getUnit(),
+        DoubleHistogramData.create(
+            isStateful() ? AggregationTemporality.CUMULATIVE : AggregationTemporality.DELTA,
+            MetricDataUtils.toDoubleHistogramPointList(
+                accumulationByLabels,
+                isStateful() ? startEpochNanos : lastCollectionEpoch,
+                epochNanos)));
+  }
+
+  @Override
+  public HistogramAccumulation accumulateDouble(double value) {
+    return HistogramAccumulation.create(
+        1, value, Collections.emptyList(), Collections.singletonList(1L));
+  }
+
+  @Override
+  public HistogramAccumulation accumulateLong(long value) {
+    return HistogramAccumulation.create(
+        1, value, Collections.emptyList(), Collections.singletonList(1L));
+  }
+
+  static final class Handle extends AggregatorHandle<HistogramAccumulation> {
+    private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+
+    @GuardedBy("lock")
+    private final State current;
+
+    Handle(double[] boundaries) {
+      current = new State(boundaries);
+    }
+
+    @Override
+    protected HistogramAccumulation doAccumulateThenReset() {
+      lock.writeLock().lock();
+      try {
+        HistogramAccumulation result =
+            HistogramAccumulation.create(
+                current.count,
+                current.sum,
+                Arrays.stream(current.boundaries).boxed().collect(Collectors.toList()),
+                Arrays.stream(current.counts).boxed().collect(Collectors.toList()));
+        current.reset();
+        return result;
+      } finally {
+        lock.writeLock().unlock();
+      }
+    }
+
+    @Override
+    protected void doRecordDouble(double value) {
+      int bucketIndex = current.findBucketIndex(value);
+
+      lock.writeLock().lock();
+      try {
+        current.record(bucketIndex, value);
+      } finally {
+        lock.writeLock().unlock();
+      }
+    }
+
+    @Override
+    protected void doRecordLong(long value) {
+      doRecordDouble((double) value);
+    }
+
+    private static final class State {
+      private long count;
+      private double sum;
+      private final double[] boundaries;
+      private final long[] counts;
+
+      public State(double[] boundaries) {
+        this.boundaries = Arrays.copyOf(boundaries, boundaries.length);
+        this.counts = new long[this.boundaries.length + 1];
+        reset();
+      }
+
+      // Benchmark shows that linear search performs better with ordinary buckets.
```

Comment:
better than what?

Reply:
better than the binary search. will update the comment.